### PR TITLE
[G-API] Expand PyParams to support constInput

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Build configuration

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

build_image:Custom=centos:7
buildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

Xbuild_image:Custom=ubuntu-openvino-2020.3.0:16.04
build_image:Custom Win=openvino-2021.3.0
build_image:Custom Mac=openvino-2021.3.0

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
// disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp [hunk 1]
```diff
@@ -22,17 +22,23 @@ namespace ie {
...
+    GAPI_WRAP PyParams() = default;
 
-    PyParams(const std::string &tag,
+    GAPI_WRAP PyParams(const std::string &tag,
              const std::string &model,
              const std::string &weights,
              const std::string &device);
 
-    PyParams(const std::string &tag,
+    GAPI_WRAP PyParams(const std::string &tag,
```
Comment (mpashchenkov) [inline modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp:32]: Alignment broken, after adding GAPI_WRAP

File: modules/gapi/src/backends/ie/bindings_ie.cpp [hunk 2]
```diff
@@ -37,3 +37,15 @@ cv::gapi::ie::PyParams cv::gapi::ie::params(const std::string &tag,
                                             const std::string &device) {
     return {tag, model, device};
 }
+
+cv::gapi::ie::PyParams& cv::gapi::ie::PyParams::constInput(const std::string &layer_name,
+                                                           const cv::Mat &data,
+                                                           TraitAs hint) {
+    m_priv->constInput(layer_name, data, hint);
```
Comment (mpashchenkov) [inline modules/gapi/src/backends/ie/bindings_ie.cpp:44]: For my information: Does Python use Generic infer?

Comment (TolyaTalamanov) [reply modules/gapi/src/backends/ie/bindings_ie.cpp:44]: Yes

**Review by dmatveev — APPROVED** [review]
**LGTM** but please consider tests for the backlog

File: modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp:32]: Done

**Review by mpashchenkov — APPROVED** [review]

File: modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp (hunk 1 above)
Comment (alalek) [reply modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp:32]: It is not done for the ctor above.
It makes sense to put "wrapping" modifiers on new line before standard declaration instead:

```.cpp
    GAPI_WRAP
    PyParams(...
```

for ctor and other methods / functions.

Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/infer/bindings_ie.hpp:32]: Done

**Comment by TolyaTalamanov** [comment]
@alalek Can it be merged ?

**Pull request merged**